src/coreclr/jit/llvm.cpp

```diff
@@ -786,7 +844,7 @@ Value* localVar(llvm::IRBuilder<>& builder, GenTreeLclVar* lclVar)
                 }
             }
 
-            _localsMap->insert({lclNum, llvmRef});
+            _localsMap->insert({lclNum, new struct LocalVar(llvmRef)});
```

Comment:
I understand we dynamically allocate these because of the desire for polymorphic behavior right now, so I have a question: do you anticipate `LocalVar` getting bigger in size? If not, I would consider making the behavior polymorphic via a descriminator union (`LocalVarLocation` or something), it would avoid the need for dynamic allocation (well, the map would still allocate dynamically, but there would one less indirection).

Also, if the dynamic allocation is there to stay, and if you are to use the compiler's allocator, it would look something like this:
```cs
// We have a global overloaded `new`
new (_compiler) LocalVar(llvmRef);
```
But for this every dynamic allocation needs to be done via the compiler's allocator.

Also not, here and elsewhere: why use `struct` explicitly? I believe C++ should be able to cope with just the type name fine...

Reply:
"why use struct explicitly" oh dear.  You've spotted by 30 year old C experience. :-)

Reply:
I don't think it needs to get bigger, this is a bit analogous to https://github.com/dotnet/runtimelab/blob/40f9ff64ae80596bcddcec16a7e1a8f57a0b2cff/src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/EvaluationStack.cs#L444 so maybe the `var_type` might be needed.  Hope not.  Will change as suggested.  Thanks